cg/meta/workflow/mutant/quality_controller/models.py

```diff
@@ -0,0 +1,103 @@
+from typing import Annotated, Any
+from pydantic import BaseModel, BeforeValidator, Field, ValidationError, ConfigDict
+from cg.store.models import Sample
+
+
+# Validators
+def empty_str_to_none(value: str) -> str | None:
+    if not isinstance(value, str):
+        raise TypeError(f"Expected a string, but got {type(value).__name__}")
+    return value or None
+
+
+def str_to_bool(value: str) -> bool:
+    if value == "TRUE":
+        return True
+    elif value == "FALSE":
+        return False
+    raise ValidationError(f"String {value} cannot be turned to bool.")
+
+
+# Models
+class SampleResults(BaseModel):
+    sample_name: str = Field(alias="Sample")
+    selection: str = Field(alias="Selection")
+    region_code: str = Field(alias="Region Code")
+    ticket: Annotated[int | None, BeforeValidator(empty_str_to_none)] = Field(alias="Ticket")
+    pct_n_bases: Annotated[float | None, BeforeValidator(empty_str_to_none)] = Field(
+        alias="%N_bases"
+    )
+    pct_10x_coverage: Annotated[float | None, BeforeValidator(empty_str_to_none)] = Field(
+        alias="%10X_coverage"
+    )
+    qc_pass: Annotated[bool, BeforeValidator(str_to_bool)] = Field(alias="QC_pass")
+    lineage: str = Field(alias="Lineage")
+    pangolin_data_version: str = Field(alias="Pangolin_data_version")
+    voc: str = Field(alias="VOC")
+    mutations: Annotated[str | None, BeforeValidator(empty_str_to_none)] = Field(alias="Mutations")
+
+
+class SamplesResultsMetrics(BaseModel):
+    samples: dict[str, SampleResults]
+
+
+class MutantPoolSamples(BaseModel):
+    samples: list[Sample]
+    external_negative_control: Sample
+    internal_negative_control: Sample
+
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+
+
+class QualityMetrics(BaseModel):
```

Comment:
From discussion: Consider renaming

Reply:
Cannot come up with better naming. Please suggest if you have any ideas!

Reply:
This model really only hold sample meta data and various types of samples as database models. The sampleResults also seem more like a metadata class than it holding results from the attributes. 

I would say calling it QualityMetrics is confusing. 
What about SampleCollection?

Reply:
With @diitaz93, decided SamplePoolAndResults!